components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/authcontext/DefaultClaimsRetrieverTest.java

```diff
@@ -0,0 +1,242 @@
+        mockStatic(OAuthComponentServiceHolder.class);
+
+        when(OAuthComponentServiceHolder.getInstance()).thenReturn(mockedOAuthComponentServiceHolder);
+        when(mockedOAuthComponentServiceHolder.getRealmService()).thenReturn(mockedRealmService);
+        mockedUserRealm = mock(UserRealm.class);
+        when(mockedRealmService.getTenantUserRealm(anyInt())).thenReturn(mockedUserRealm);
+        when(mockedUserRealm.getUserStoreManager()).thenReturn(mockedUserStoreManager);
+        Map<String, String> expectedMappingTrue = new HashMap<>();
+        when(mockedUserStoreManager.getUserClaimValue("user", "[\"https://www.wso2.org/address\", \"https://www.wso2.org/email\"]", "read")).thenReturn(expectedMappingTrue.toString());
+
+        String[] claims = {"https://www.wso2.org/address", "https://www.wso2.org/email"};
+
+        assertNotNull(defaultClaimsRetriever.getClaims("admin", claims));
+    }
+
+    @Test(expectedExceptions = IdentityOAuth2Exception.class)
+    public void testGetClaims1() throws Exception {
+        SortedMap<String, String> claimValues;
+        int tenantId = MultitenantConstants.SUPER_TENANT_ID;
+
+        mockStatic(OAuthServerConfiguration.class);
+        when(OAuthServerConfiguration.getInstance()).thenReturn(mockedOAuthServerConfiguration);
+        when(mockedOAuthServerConfiguration.getTimeStampSkewInSeconds()).thenReturn((long) 3);
+        mockStatic(OAuth2Util.class);
+        mockStatic(MultitenantUtils.class);
+        mockStatic(OAuthComponentServiceHolder.class);
+
+        when(OAuthComponentServiceHolder.getInstance()).thenReturn(mockedOAuthComponentServiceHolder);
+        when(mockedOAuthComponentServiceHolder.getRealmService()).thenReturn(mockedRealmService);
+        mockedUserRealm = mock(UserRealm.class);
+        when(mockedRealmService.getTenantUserRealm(anyInt())).thenReturn(mockedUserRealm);
+        when(mockedUserRealm.getUserStoreManager()).thenThrow(new UserStoreException("UserStoreException"));
+        Map<String, String> expectedMappingTrue = new HashMap<>();
+        when(mockedUserStoreManager.getUserClaimValue(anyString(), anyString(), anyString())).thenThrow(new UserStoreException("UserStoreException"));
+
+        String[] claims = {"https://www.wso2.org/address", "https://www.wso2.org/email"};
+
+        assertNotNull(defaultClaimsRetriever.getClaims("admin", claims));
+    }
+
+    @Test
+    public void testGetDefaultClaims() throws Exception {
+
+        mockStatic(OAuthServerConfiguration.class);
+        when(OAuthServerConfiguration.getInstance()).thenReturn(mockedOAuthServerConfiguration);
+        when(mockedOAuthServerConfiguration.getTimeStampSkewInSeconds()).thenReturn((long) 7);
+        mockStatic(OAuth2Util.class);
+        mockStatic(MultitenantUtils.class);
+        mockStatic(OAuthComponentServiceHolder.class);
+
+        when(OAuthComponentServiceHolder.getInstance()).thenReturn(mockedOAuthComponentServiceHolder);
```

Comment:
Why did you have to mock realm service?
Is it used internally by the method tested?

Reply:
Yes. mocked realm service returns an object of User realm.